### PR TITLE
New serverless pattern -  Firehose Transformation pattern (CDK + Typescript)

### PR DESCRIPTION
**Pull request by ejschaefer**
*Issue #, if available:*

*Description of changes:*
New Firehose Transformation pattern (CDK + Typescript)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bfreiberg** [comment]
Hi, thanks for your contribution. There are two issues with your `example-pattern.json`:
1. `language` is case-sensitive, please use `TypeScript`
2. please only use the file name for `gitHub.template.templateFile`

**Comment by bfreiberg** [comment]
And please create an issue that links to this request, as explained in step 3 here https://github.com/aws-samples/serverless-patterns/blob/main/PUBLISHING.md

File: firehose-transformation-cdk-typescript/src/README.md
Comment (bfreiberg) [inline firehose-transformation-cdk-typescript/src/README.md:1]: At least the first paragraph is not really helpful/generic. Please include at least a one-liner with some context

File: firehose-transformation-cdk-typescript/src/jest.config.js
Comment (bfreiberg) [inline firehose-transformation-cdk-typescript/src/jest.config.js:1]: Is that file really necessary if there are no tests anyway?

File: firehose-transformation-cdk-typescript/src/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "build": "tsc",
+    "watch": "tsc -w",
+    "test": "jest",
+    "cdk": "cdk"
+  },
+  "devDependencies": {
+    "@types/jest": "^29.5.5",
+    "@types/node": "20.7.1",
+    "aws-cdk": "2.102.0",
+    "jest": "^29.7.0",
```
Comment (bfreiberg) [inline firehose-transformation-cdk-typescript/src/package.json:17]: see comment about tests above

File: firehose-transformation-cdk-typescript/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## How it works
+
+Kinesis Data Firehose can invoke a Lambda function to transform incoming source data and deliver the transformed data to destinations. In this architecture, Kinesis Data Firehose invokes the specified Lambda function asynchronously with each buffered batch using the AWS Lambda synchronous invocation mode. The transformed data is sent from Lambda to Kinesis Data Firehose. Kinesis Data Firehose then sends it to the destination S3 bucket when the specified destination buffering size or buffering interval is reached, whichever happens first.
+
+## Testing
+
+1. Open the Kinesis Data Firehose console at https://console.aws.amazon.com/firehose/
+
+2. Choose the {stack-name}-firehoststream-{stream-id} delivery stream
```
Comment (bfreiberg) [inline firehose-transformation-cdk-typescript/README.md:47]: When I open the console there is a very visible warning about logging to CloudWatch Logs not being enabled. It's not necessary for this pattern to enable it but consider adding a short sentence about the trade-off between conciseness for demo purposes and best practices somewhere in the README.

**Comment by bfreiberg** [comment]
Hi @ejschaefer , did you get a chance to look at my comments yet?

**Comment by ejschaefer** [comment]
Hi Ben,

Yes, I’m back from PTO today. I made some changes locally including your
recommendations. I’ll validate and commit today.

-Ed

File: firehose-transformation-cdk-typescript/example-pattern.json [hunk 3]
```diff
@@ -0,0 +1,65 @@
+{
+  "title": "Kinesis Firehose Data Transformation with Lambda",
+  "description": "Transform incoming source data and deliver the transformed data to destinations.",
```
[the remark this answers is not in the document]
Comment (ejschaefer) [reply firehose-transformation-cdk-typescript/example-pattern.json]: fixed

File: firehose-transformation-cdk-typescript/src/README.md
Comment (ejschaefer) [reply firehose-transformation-cdk-typescript/src/README.md:1]: removed this extra/generic README that CDK created

File: firehose-transformation-cdk-typescript/src/jest.config.js
Comment (ejschaefer) [reply firehose-transformation-cdk-typescript/src/jest.config.js:1]: removed jest

File: firehose-transformation-cdk-typescript/src/package.json (hunk 1 above)
Comment (ejschaefer) [reply firehose-transformation-cdk-typescript/src/package.json]: removed jest

File: firehose-transformation-cdk-typescript/README.md (hunk 2 above)
Comment (ejschaefer) [reply firehose-transformation-cdk-typescript/README.md]: added CW logs to destination and backup configs.

**Comment by bfreiberg** [comment]
Looks good, thanks for your contribution.

**Comment by bls20AWS** [comment]
live: https://serverlessland.com/patterns/firehose-transformation-cdk-typescript
